DailyKit/CallManager/CallManager.swift

```diff
@@ -0,0 +1,134 @@
+    public var url: URL? { subjects.url.value }
+    public var callState: CallState { subjects.callState.value }
+    public var camera: CallCamera { subjects.camera.value }
+    public var microphone: CallMicrophone { subjects.microphone.value }
+    public var participants: CallParticipants { subjects.participants.value }
+
+    private let callClient: CallClient
+    private let subjects: CallClientSubjects
+    private var participantsBuilder: CallParticipants.Builder
+
+    public convenience init() {
+        self.init(callClient: CallClient())
+    }
+
+    required init(callClient: CallClient) {
+        self.callClient = callClient
+        self.participantsBuilder = CallParticipants.Builder(
+            local: callClient.participants.local.asCallParticipant
+        )
+        self.subjects = CallClientSubjects(
+            url: callClient.url,
+            callState: callClient.callState,
+            camera: CallCamera(callClient.inputs.camera),
+            microphone: CallMicrophone(callClient.inputs.microphone),
+            participants: participantsBuilder.build()
+        )
+
+        callClient.delegate = self
+
+        applyDefaults()
+    }
+
+    private func applyDefaults() {
+        callClient.setInputsEnabled([
+            .camera: true,
+            .microphone: false
+        ])
+
+        callClient.setIsPublishing([
+            .camera: true,
+            .microphone: true
+        ])
+    }
+
+    public func publisher<T>(for property: CallClientProperty<T>) -> AnyPublisher<T, Never> {
+        subjects.publisher(for: property)
+    }
+
+    public func toggleCamera(_ camera: CallCamera) {
+        switch camera.video {
+        case .muted:
+            callClient.setInputsEnabled([.camera: true])
+        case .unmuted:
+            callClient.setInputsEnabled([.camera: false])
+        }
+    }
+
+    public func toggleMicrophone(_ microphone: CallMicrophone) {
+        switch microphone.audio {
+        case .muted:
+            callClient.setInputsEnabled([.microphone: true])
+        case .unmuted:
+            callClient.setInputsEnabled([.microphone: false])
+        }
+    }
+
+    public func flipCamera(_ camera: CallCamera) {
+        switch camera.mode {
+        case .user:
+            callClient.updateInputs(.set(camera: .set(settings: .set(facingMode: .set(.environment)))))
+        case .environment:
+            callClient.updateInputs(.set(camera: .set(settings: .set(facingMode: .set(.user)))))
+        }
+    }
+
+    public func setUsername(_ username: String?) {
+        callClient.set(username: username)
+    }
+
+    public func join(url: URL) {
+        callClient.join(url: url)
+        subjects.url.send(url)
+
+        // TODO: Find a cleaner way to handle the joining transition.
+        // Send a `joining` value, so the UI will immediately transition to the waiting screen.
+        subjects.callState.send(.joining)
```

Comment:
Potentially long enough to need an explicit loading state. This is an optimization that I made in response to some feedback from someone testing the app in a location with poor network connectivity. I want to see if there's anything we can do in the SDK to update the state faster in those conditions. Ideally we can transition to `joining` immediately and remove this part, otherwise I think we'll want to introduce a UI-specific call state type that can represent both the `CallState` and user intent. `CallCallState`? 🙃